docs/further_useful_features.rst

```diff
@@ -0,0 +1,438 @@
+            "I": np.array([9] * 4, dtype="int32"),
+            "J": pd.Series(3, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190604"),
+            "L": 2.,
+        }
+    )
+    df2
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "another_unique_dataset_identifier",
+        dfs = [
+            {
+                "data":
+                {
+                    "table1": df,
+                    "table2": df2
+                }
+            }
+        ]
+    )
+    dm.tables
+    dm.partitions
+
+
+Below is an example where we update the existing dataset ``another_unique_dataset_identifier``
+with new data for ``table1`` and ``table2``:
+
+.. ipython:: python
+
+    another_df2 = pd.DataFrame(
+        {
+            "G": "bar",
+            "H": pd.Categorical(["prod", "dev", "prod", "dev"]),
+            "I": np.array([12] * 4, dtype="int32"),
+            "J": pd.Series(4, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190614"),
+            "L": 10.,
+        }
+    )
+    another_df2
+
+    dm = update_dataset_from_dataframes(
+        {
+            "data":
+            {
+                "table1": another_df,
+                "table2": another_df2
+            }
+        },
+        store=store_factory,
+        dataset_uuid="another_unique_dataset_identifier"
+    )
+    dm.tables
+    dm.partitions
+
+
+Trying to update only a subset of tables throws a ``ValueError``:
+
+.. ipython::
+
+    @verbatim
+    In [45]: update_dataset_from_dataframes(
+       ....:        {
+       ....:           "data":
+       ....:           {
+       ....:              "table2": another_df2
+       ....:           }
+       ....:        },
+       ....:        store=store_factory,
+       ....:        dataset_uuid="another_unique_dataset_identifier"
+       ....:        )
+       ....:
+    ---------------------------------------------------------------------------
+    ValueError: Input partitions for update have different tables than dataset:
+    Input partition tables: {'table2'}
+    Tables of existing dataset: ['table1', 'table2']
+
+
+Adding data to existing datasets is not the only functionality achievable within an update
+and it can even be used to remove or replace data. To do this we use the ``delete_scope``
+keyword argument as shown in the example below:
+
+.. ipython:: python
+
+    dm = update_dataset_from_dataframes(
+        None,
+        store=store_factory,
+        dataset_uuid="partitioned_dataset",
+        partition_on="E",
+        delete_scope=[{"E": "train"}]
+    )
+    dm.partitions
+
+
+As we can see, we specified using a dictionary that data where the column ``E`` has the
+value ``train`` should be removed. Looking at the partitions after the update, we see that
+the partition ``E=train`` has been removed.
+
+.. note:: We defined ``delete_scope`` over a value of ``E``. ``E`` also happens to be a
```

Comment:
_happens_ implies that it is by accident. There are two very important things in here:
1. delete scope *only* works on indexed columns
2. it *should* only be used on partitioned columns due to the 1:1 mapping I mentioned before. Without the guarantee of 1:1 we would accidentally remove data with different values